compiler/src/dmd/e2ir.d

```diff
@@ -1819,6 +1819,57 @@ elem* toElem(Expression e, IRState *irs)
             // to object.__cmp in druntime.
             assert(0);
         }
+        else if (t1.ty == Tvector)
+        {
+            /* Rewrite in terms of < operator
+             */
+            bool swap;       // swap operands
+            bool comp;       // complement result
+            switch (eop)
+            {
+                case OPgt:                           break; //   x > y
+                case OPlt: swap = true;              break; //   y > x
+                case OPle:              comp = true; break; // !(x > y)
+                case OPge: swap = true; comp = true; break; // !(y > x)
+                default:   assert(0);
+            }
+
+            elem* e1 = toElem(ce.e1, irs);
+            elem* e2 = toElem(ce.e2, irs);
+
+            elem* ex;
+            if (swap)
+            {
+                // put side effects of e1 into ex
+                if (el_sideeffect(e1) && e2.Eoper != OPconst)
```

Comment:
There's no test for comparisons with side effects in the operands

Reply:
there is now